File: packages/types/src/meeting.ts
export interface ActionItem {
  id: string
  description: string
  responsible: string
  dueDate: string
}

export interface AgendaItem {
  id: string
  title: string
  description: string
  content?: string
  actionItems?: ActionItem[]
  fixed?: boolean
}

export interface MeetingType {
  id: string
  code: string
  name: string
}

export interface User {
  name?: string | null
  email: string
}

export interface Meeting {
  id: string
  identifier: string
  date: string
  startTime?: string | null
  endTime?: string | null
  location?: string | null
  agenda: Record<string, unknown>
  content: string
  actionItems?: Record<string, unknown>[] | null
  createdAt: string
  updatedAt?: string
  meetingType: MeetingType
  createdBy: User
}

export interface MeetingResponse {
  meetings: Meeting[]
  pagination: {
    page: number
    limit: number
    total: number
    totalPages: number
  }
}

File: api/app/schemas/meeting.py
from typing import Any

from app.schemas.base import ORMModel, PrismaDateTime
from app.schemas.document import CreatedByOut
from app.schemas.meeting_type import MeetingTypeOut


class AttendeeProfileOut(ORMModel):
    name: str | None
    email: str


class MeetingAttendeeOut(ORMModel):
    id: str
    present: bool
    role: str | None
    created_at: PrismaDateTime
    meeting_id: str
    profile_id: str
    profile: AttendeeProfileOut


class MeetingBareOut(ORMModel):
    """Bare meeting row (no relations) — the POST create response shape."""

    id: str
    identifier: str
    date: PrismaDateTime
    start_time: str | None
    end_time: str | None
    location: str | None
    agenda: Any
    content: str
    decisions: Any | None
    action_items: Any | None
    created_at: PrismaDateTime
    updated_at: PrismaDateTime
    meeting_type_id: str
    created_by_id: str


class MeetingOut(MeetingBareOut):
    meeting_type: MeetingTypeOut
    created_by: CreatedByOut
    attendees: list[MeetingAttendeeOut]


class MeetingSearchListOut(MeetingBareOut):
    """Search fallback shape: meeting + meetingType + createdBy, no attendees."""

    meeting_type: MeetingTypeOut
    created_by: CreatedByOut
